### DISCUSSION TITLE
Be able to set FilamentView::registerRenderHook per panel

### DISCUSSION THREAD

**Discussion by miguilimzero** [filamentphp/filament#7623] [Ideas]
So I currently have an application with two panels, the user app panel and the administration panel, and I wanted to customize the Filament top bar. Following the documentation, I need to set `FilamentView::registerRenderHook` for `panels::topbar.start` name. The documentation says that I mutis put this code in some service provider.

The problem comes because customization is applied for all panels. I tried to put the `registerRenderHook` in the specific `PanelProvider` boot() function I want to be applied to, but even while doing so, the customized part is still rendered in all panels.

**Comment by danharrin** [comment]
https://filamentphp.com/docs/3.x/panels/configuration#render-hooks

**Comment by sbc640964** [comment]
@danharrin answer is the correct one specifically for your question.
In any case you can always check the current panel
```
filament()->getCurrentPanel()
```
